Assets/Scripts/Score/HighScoreHandler.cs

```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+using TMPro;
+public class HighScoreHandler : MonoBehaviour , ScoreSystem
+{
+    [Header("Text elements")]
+    [SerializeField] private TextMeshProUGUI _highscoreText;
+    [SerializeField] private TextMeshProUGUI _creditText;
+
+    private int _highscore;
+    private int _credits;
+
+    public static HighScoreHandler _highScoreHandler;
+    private void Awake()
+    {
+        _highScoreHandler = this;
+    }
+    public void OnGameOver(int score)
+    {
+        if (score >= _highscore)
+            _highscore = score;
+
+        _credits += CreditsConversion(score);
+    }
```

Comment:
```suggestion
     private void Awake()
    {
        //subscribe to event
    }
```
hier kunnen we hebben dat we een OnValueChange event hebben die word gefired als punten worden toegevoegd. dan kunnen andere systemen kijken naar de nieuwe value zonder tight coupling